grasa_event_locator/views.py

```diff
@@ -32,7 +32,16 @@ def admin(request):
                 pendingEditList = Program.objects.filter(isPending=True).exclude(editOf=0)
                 context = {'pendingUserList' : pendingUserList, 'pendingEventList' : pendingEventList, 'pendingEditList' : pendingEditList}
                 if request.method == "POST":
-                        change_username(request.user.username, request.POST['changeemail'] , request)
+                        if request.POST.get('changeemail'):
+                            change_username(request.user.username, request.POST['changeemail'] , request)
+                        if request.POST.get('current') and (request.POST.get('new') == request.POST.get('confirm')):
```

Comment:
What real-world criteria distinguishes between the two different `POST` metadata types? I know it might be nitpicky, but please add a one-line comment above each `if` conditional to briefly explain the trigger criteria expected for one conditional over the other.

Does this make sense?

Reply:
I'll add a comment in-code, but will give an explanation here.

Basically, the admin page (provider page follows a similar train of thought) has two functions: either change the email address of the account, or change the password of the account. The first if statement changes the email address (which we're storing as the username). The second if statement runs through the password change process. If the two functions were together without the if statement, there would be errors indicating, "oh, one of the fields (the one that doesn't provide input to the intended function) is empty". And even if if had data sent, it would, in this case, try to both change the email using that, and then try to change the password, hence the need to determine whether the field contains any data. If it does, then perform it's function (in this case, changing the username), if not, skip it over and check the next field. If it has data (in this case current/new/confirm password), perform it's function (change the password).

The comments should have some more context.